src/Confluent.Kafka/ProducerBuilder.cs

```diff
@@ -0,0 +1,292 @@
+        {
+            this.StatisticsHandler = statisticsHandler;
+            return this;
+        }
+
+        /// <summary>
+        ///     Set the handler to call on error events e.g. connection failures or all
+        ///     brokers down. Note that the client will try to automatically recover from
+        ///     errors that are not marked as fatal - such errors should be interpreted
+        ///     as informational rather than catastrophic.
+        /// </summary>
+        /// <remarks>
+        ///     Executes on the poll thread (by default, a background thread managed by
+        ///     the producer).
+        /// </remarks>
+        public ProducerBuilder<TKey, TValue> SetErrorHandler(Action<Producer<TKey, TValue>, Error> errorHandler)
+        {
+            this.ErrorHandler = errorHandler;
+            return this;
+        }
+
+        /// <summary>
+        ///     Set the handler to call when there is information available
+        ///     to be logged. If not specified, a default callback that writes
+        ///     to stderr will be used.
+        /// </summary>
+        /// <remarks>
+        ///     By default not many log messages are generated.
+        ///
+        ///     For more verbose logging, specify one or more debug contexts
+        ///     using the 'debug' configuration property. The 'log_level'
+        ///     configuration property is also relevant, however logging is
+        ///     verbose by default given a debug context has been specified,
+        ///     so you typically shouldn't adjust this value.
+        ///
+        ///     Warning: Log handlers are called spontaneously from internal
+        ///     librdkafka threads and the application must not call any
+        ///     Confluent.Kafka APIs from within a log handler or perform any
+        ///     prolonged operations.
+        /// </remarks>
+        public ProducerBuilder<TKey, TValue> SetLogHandler(Action<Producer<TKey, TValue>, LogMessage> logHandler)
+        {
+            this.LogHandler = logHandler;
+            return this;
+        }
+
+        /// <summary>
+        ///     The serializer to use to serialize keys.
+        /// </summary>
+        public ProducerBuilder<TKey, TValue> SetKeySerializer(ISerializer<TKey> serializer)
+        {
+            this.KeySerializer = serializer;
+            return this;
+        }
+
+        /// <summary>
+        ///     The serializer to use to serialize values.
+        /// </summary>
+        public ProducerBuilder<TKey, TValue> SetValueSerializer(ISerializer<TValue> serializer)
+        {
+            this.ValueSerializer = serializer;
+            return this;
+        }
+
+        /// <summary>
+        ///     The serializer to use to serialize keys.
+        /// </summary>
+        public ProducerBuilder<TKey, TValue> SetKeySerializer(IAsyncSerializer<TKey> serializer)
+        {
+            this.AsyncKeySerializer = serializer;
```

Comment:
what happens if you set both KeySerializer and AsyncKeySerializer? Seems like that should not be allowed.